### ISSUE TITLE
"Close Other Panes" seems to close other tabs

### ISSUE THREAD

**Issue by huyz** [luckman212/obsidian-split-pane-view#1]
The "Close Other Panes" closes most but not all my other tabs for some reason. I would expect split panes to be closed, not other tabs inside the tab group of my current file.

**Comment by luckman212** [comment]
Are you running [v1.0.3](https://github.com/luckman212/obsidian-split-pane-view/releases/tag/1.0.3)? If not, please download it and test again.

Can you post screenshots of "before" and "after" showing the panes that did not close? And also your console output if there is any?

**Comment by huyz** [comment]
I'm running whatever's in master.  I think it's v1.0.2 but manifest.json wasn't updated so I'm not 100% sure.

Before:
![screenshot 2022-11-19T070625Z](https://user-images.githubusercontent.com/128394/202839252-f23dbdee-cf85-4328-a051-b5314f3c5db4.png)

After:
![screenshot 2022-11-19T070634Z](https://user-images.githubusercontent.com/128394/202839249-4d51db1c-5d2e-446c-9c82-32a7abc362cf.png)

My other tab got nuked.

**Comment by luckman212** [comment]
Thanks for letting me know about the manifest- yes I forgot to update it. I'll fix shortly.

As for the actual behavior, that actually looks correct to me. The intention of the _Close Other_ function is to close all other leaves, without regard for whether they are tabs or separate panes.

If you want stacked tabs within the currently active pane to remain open, I will need to make a setting for that.

**Comment by huyz** [comment]
Not sure why you're calling this "stacked tabs".  To me those are regular tabs.
These are stacked tabs:
![screenshot 2022-11-19T163704Z](https://user-images.githubusercontent.com/128394/202861651-d62ac3ca-ff50-46c8-bb24-bb45d20ff0aa.png)

You really think the current behavior makes sense?  All I want to do is create a split and the plugin kills all the other tabs.  What do tabs have to do with splits?  I mean, sure, create a setting—that'll work for me.  But don't you think the current behavior is surprising and unhelpful to users?

**Comment by luckman212** [comment]
I used the wrong terminology for stacked tabs. I meant "tabs that are grouped together into a single pane" (leaf). In your case, if you want to activate the side-by-side view _without_ closing other leaves, does turning off the toggle below not do what you want?

![screenshot_8ub6iLfQ](https://user-images.githubusercontent.com/1992842/202863845-c2fb6675-7c0c-4653-9222-2834e9e78d48.png)

Obviously I don't think the behavior is "surprising and unhelpful" 🙀 otherwise I wouldn't have published it. But, I am open to feedback from you and others on how to better implement this or re-word the descriptions so there are no surprises. I supposed I should change the word _Panes_ to _Leaves_ to be more explicit.

One thing I could probably implement is, when that toggle is off, we could iterate all the leaves and only close leaves that are part of the current grouping. In that way, repeated pressing of the hotkey would not open 2,3,4...duplicate tabs—it would simply reset the current split view for the active tab without interfering with other leaves.

**Comment by huyz** [comment]
I just want "splits" (aka "split panes" or simply "panes") closed, not "tabs". "Leaves" is not a common term—never seen it, so I assume that must come from the Obsidian API. Maybe you mean "Close other files"?

> repeated pressing of the hotkey would not open 2,3,4...duplicate tabs

Not sure what you mean here.  I thought this plugin only opened a split pane. How can there be duplicate tabs.

I feel like I have a completely wrong model of what this plugin is trying to do.

**Comment by luckman212** [comment]
Yes, leaves are what the views are called internally (API). Yes, I think maybe you are looking for a different feature entirely here. Are you saying you want a feature to close all splits (sort of a "combine all open splits/views into a single pane")? That is definitely not the purpose of this plugin. If I am misunderstanding, please manually arrange your layout "before" and "after" and paste screenshots here so it's clear.

**Comment by huyz** [comment]
What my friend and I miss from before Obsidian v1 is this: always having 2 splits, one for editing view and one for reading view.

The problem is that when tabs were added in v1, the "linked" reading view on the right doesn't change when one switches tabs in the left split.  To me this is a UX flaw, and there are feature requests for Core to change this, but who knows if and when they'll get to it.
I thought your plugin was going to provide a nice workaround: the ability for me to, once i switched to a new tab, hit a command to quickly open a split reading view; and because i only have so much horizontal real estate, I would want all existing splits on the right to be closed to make room.
Does this make sense?

**Comment by luckman212** [comment]
I keep asking you for screenshots... _please post screenshots_ of what you are asking for. Before and After. I will see if this is something that can be done.

I think you are saying, if you have a random arrangement of tabs, panes, splits, etc, you want the plugin to collapse ALL of them to the "left" and then open a single split pane view to the right of just the active tab. Is that right?

**Comment by huyz** [comment]
So if I start out with:
![screenshot 2022-11-19T194310Z](https://user-images.githubusercontent.com/128394/202868750-4d8f8cb5-8511-4076-a94a-b3029da7ab6b.png)

If I use core functionality by ⌘-clicking this button:
![screenshot 2022-11-19T194402Z](https://user-images.githubusercontent.com/128394/202868806-b2974ecf-4b88-4764-8c89-2f3a4020f8fb.png)
I get:
![screenshot 2022-11-19T194414Z](https://user-images.githubusercontent.com/128394/202868814-e06dfea4-9791-408c-b31e-2c61b0a50671.png)


With your plugin and "Close other panes" set to on, I would get:
![screenshot 2022-11-19T194547Z](https://user-images.githubusercontent.com/128394/202868765-6d2a36ab-d9e4-49e6-9d96-28d7ab56a240.png)


What I was expecting, and what I was asking for is:
![screenshot 2022-11-19T194439Z](https://user-images.githubusercontent.com/128394/202868861-fbdbb33f-df8c-411c-bde2-050d8821ed0b.png)

But actually, to me this would be ideal as I would lose no tabs:
![screenshot 2022-11-19T195516Z](https://user-images.githubusercontent.com/128394/202869094-03af936d-0c79-49d3-822b-392f61a431e0.png)

Now, if there happened to be more than two split panes to begin with, then it would be ok for the third, fourth, etc. splits to be closed, as that would free up horizontal real estate for the reading view.

In any of the above solutions, it would be ideal if when the plugin closed any split, any of the tabs in that split were shifted into the right-most split.  But if that's harder to implement, then I can live without that, and losing the tabs in the closed splits would be fine for me (the reason is that I tend to have ephemeral notes in my non-main split pane; of course, other users may have different schemes).

**Comment by luckman212** [comment]
Screenshots were helpful. I think I understand what you're after now. Basically, preserve tabs you have open no matter what, but always set up the view as a single-split with the active file on the left and right.

That's a challenge, but I will see if I can figure out a way to get it done. It will take some time, I'm open to PRs if you have any ideas on how to implement it.

**Comment by huyz** [comment]
Thanks for considering it.  I'll try to take a look at it once I'm done looking at the Obsidian Plugin Manager.
